runtests.py

```diff
@@ -328,7 +328,7 @@ def build_project(args):
     # Always use ccache, if installed
     env['PATH'] = os.pathsep.join(EXTRA_PATH + env.get('PATH', '').split(os.pathsep))
     cvars = distutils.sysconfig.get_config_vars()
-    if 'gcc' in cvars.get('CC', ''):
+    if 'gnu-gcc' in cvars.get('CC', ''):
```

Comment:
this code path is not macOS specific, so shouldn't you check for both `'gcc'` and `'gnu-gcc'`?

Reply:
On OSX with Xcode, `CC=gcc`. Therefore, `'gcc' in cvars.get('CC', '')` is true. I think we want to check specifically for `'gnu-gcc'` only. I might be missing your point.

Reply:
clang on OSX reports itself as `gcc`, but then does not support some of the warning flags we add, resulting in a failed build.

Reply:
Okay, my suggestion needs a tweak. What I meant is: this seems to break things for GCC on other platforms. So something like:

    if 'gcc' in cvars.get('CC', ''):
        if sys.platform == 'darwin' and 'gnu-gcc' in cvars.get('CC', ''):
            # Clang identifies itself as GCC, so check for the actual (GNU) GCC
            etc.

Reply:
Sensible. I chose slightly different logic that expresses this same idea. Force-pushed for review.